proto/src/determined/api/v1/experiment.proto

```diff
@@ -675,3 +676,30 @@ message PostSearcherOperationsRequest {
 
 // Response to PostSearcherOperationsResponse.
 message PostSearcherOperationsResponse {}
+
+// Request for searching experiments
+message SearchExperimentsRequest {
+  // ID of the project to look at
+  int32 project_id = 1;
+  // How many experiments to skip before including in the results
+  int32 offset = 2;
+  // How many results to show
+  int32 limit = 3;
+}
+
+// combination of experiment and best trial with metrics
+message SearchExperimentExperiment {
+  determined.experiment.v1.Experiment experiment = 1;
+  determined.trial.v1.Trial best_trial           = 2;
```

Comment:
i don't think `trial.v1.Trial` has metrics. based on the conversation with alex, we can do a [`checkpoint.v1.Checkpoint`](https://github.com/determined-ai/determined/blob/9bb9caa80928912b940fb5eca0fb45462058861a/proto/src/determined/checkpoint/v1/checkpoint.proto#L48-L72)? (metrics are in `CheckpointTrainingMetadata`). 

Alternatively, in order to avoid having to either make multiple queries (now up to three if we're doing checkpoints), or doing a join and then reconstituting the result into the separate structures, my preference would just to make a message from scratch that's tailored to what we actually want to send back, like [this one](https://github.com/determined-ai/determined/blob/9bb9caa80928912b940fb5eca0fb45462058861a/proto/src/determined/api/v1/trial.proto#L163). 


Since it already has almost all the experiment + trial + metrics fields we want, we can even just copy that one and call it `AugmentedExperiments`, and also reuse/copy the existing [code to convert](https://github.com/determined-ai/determined/blob/9bb9caa80928912b940fb5eca0fb45462058861a/master/internal/trials/postgres_trials.go#L47) from bun structs to api struct. 

I don't think the duplication is a big deal, since all the `AugmentedTrial` related code will be getting cleaned up in the near future anyway.